CHANGELOG.md

```diff
@@ -3,7 +3,64 @@
+- Update downloader's Etag to value of the last successful activation
+
+### Tooling
+
+- bundle, cmd/build: Do not write manifest if empty ([#3480](https://github.com/open-policy-agent/opa/issues/3480))
+- bundle: Compare manifest metadata in Equal() function
+- cmd/fmt: fix incorrect help text ([#3518](https://github.com/open-policy-agent/opa/issues/3518)) authored by @[andrehaland](https://github.com/andrehaland)
+- cmd: Do not print nil errors in bench subcommand ([#3530](https://github.com/open-policy-agent/opa/issues/3530))
+
+### Rego
+
+- rego: Expose random seeding in rego package ([#3560](https://github.com/open-policy-agent/opa/issues/3560))
+- topdown: Fix PE unknown check to avoid saving unnecessarily ([#3552](https://github.com/open-policy-agent/opa/issues/3552))
+- topdown: Fix inlining controls for functions ([#3463](https://github.com/open-policy-agent/opa/issues/3463))
+- topdown: (shallow) partial eval of ref to empty collection in presence of `with` statement ([#3420](https://github.com/open-policy-agent/opa/issues/3420))
+- ast, topdown: Improve indexer to understand function args
+- topdown/cache: Check cache value size during insert operation
+- ast: Improve InterfaceToValue to handle non-native types
+- types: Use binary search for static property lookup
+- topdown: Fix `indexof` when using UTF-8 characters
+- topdown: Fix `http.send` flaky test
+
+#### Wasm
+
+- SDK: update wasmtime-go to 0.28.0, authored by @[olivierlemasle](https://github.com/olivierlemasle)
+- Bugfix: count() now counts invalid UTF-8 runes (previously aborted)
+- Compiler: emit unreachable instruction after opa_abort()
+
+### Miscellaneous
+
+- Build: Bump Go version to 1.16.3 ([#3555](https://github.com/open-policy-agent/opa/issues/3555))
+- `make check` now uses golangci-lint via docker, authored by @[willbeason](https://github.com/willbeason)
+- The statically-built linux binary is properly used in the make targets that need it, and published to edge binaries.
+- Built binaries are now smoke tested on Windows, macos, and Linux.
+- internal/gojsonschema: fix test failing with Go 1.17 rc ([#3589](https://github.com/open-policy-agent/opa/issues/3589)) authored by @[olivierlemasle](https://github.com/olivierlemasle)
+- ci: enable dependabot for wasmtime-go
+
+#### Documentation
+
+- OAuth2/OIDC: Fixed concat command in metadata discovery method ([#3543](https://github.com/open-policy-agent/opa/pull/3543), @[iggbom](https://github.com/iggbom))
+- Policy Reference: syntax highlighting EBNF grammar (@[PatMyron](https://github.com/PatMyron))
+- Extending OPA: fix typo (@[dxps](https://github.com/dxps))
+- Extending OPA: marshal the decision log (@[TheLunaticScripter](https://github.com/TheLunaticScripter))
+- Kubernetes Introduction: fix typo (@[dbaker-rh](https://github.com/dbaker-rh))
+- Envoy: Add guidance for OPA-Envoy benchmarks
+- Envoy: update opa-envoy-plugin and contrib default branch
```

Comment:
Nit: These refer to changes in the `opa-envoy-plugin` and `contrib` repos..correct ? Have we includes changes outside of OPA proper in the OPA release notes before ?

Reply:
These are docs references to those repos. I don't suppose it's too important to mention them here, let's drop them.